pkg/provider/aws/auth/auth.go

```diff
@@ -122,15 +122,26 @@ func New(ctx context.Context, store esv1beta1.GenericStore, kube client.Client,
 	}
 
 	for _, aRole := range prov.AdditionalRoles {
+		// TODO: delete this comment later
+		// assumeRoler calls AssumeRoleFunc, which is where
+		// the assertion for the test lives and currently fails
 		stsclient := assumeRoler(sess)
 		sess.Config.WithCredentials(stscreds.NewCredentialsWithClient(stsclient, aRole))
 	}
 
+	sess_ext_id := prov.ExternalID
```

Comment:
We still need to change this variable to something like `sessExtID`
```suggestion
	sessExtID := prov.ExternalID
```